tt_metal/hw/inc/debug/sanitize_noc.h

```diff
@@ -34,49 +34,52 @@ typedef bool debug_sanitize_noc_dir_t;
 #define DEBUG_SANITIZE_NOC_UNICAST false
 typedef bool debug_sanitize_noc_cast_t;
 
-#define DEBUG_VALID_L1_ADDR(a, l, extra_alignment_bytes) \
+// TODO(PGK): remove soft reset when fw is downloaded at init
+#define DEBUG_VALID_REG_ADDR(a, l) \
     ( \
-        (a >= MEM_L1_BASE) && \
-        (a + l <= MEM_L1_BASE + MEM_L1_SIZE) && \
-        ((a) + (l) > (a)) && \
-        ((a % NOC_L1_ALIGNMENT_BYTES) == 0) && \
-        ((a % extra_alignment_bytes) == 0) \
+        ( \
+            ( \
+                ((a) >= NOC_OVERLAY_START_ADDR) && \
+                ((a) < NOC_OVERLAY_START_ADDR + NOC_STREAM_REG_SPACE_SIZE * NOC_NUM_STREAMS) \
+            ) || \
+            ((a) == RISCV_DEBUG_REG_SOFT_RESET_0) \
+        ) \
+        && (l) == 4 \
     )
-
-// TODO(PGK): remove soft reset when fw is downloaded at init
-#define DEBUG_VALID_REG_ADDR(a)                                                        \
-    ((((a) >= NOC_OVERLAY_START_ADDR) &&                                               \
-      ((a) < NOC_OVERLAY_START_ADDR + NOC_STREAM_REG_SPACE_SIZE * NOC_NUM_STREAMS)) || \
-     ((a) == RISCV_DEBUG_REG_SOFT_RESET_0))
-#define DEBUG_VALID_WORKER_ADDR(a, l, extra_alignment_bytes) \
+#define DEBUG_VALID_WORKER_ADDR(a, l) \
     ( \
-        DEBUG_VALID_L1_ADDR(a, l, extra_alignment_bytes) || \
-        (DEBUG_VALID_REG_ADDR(a) && (l) == 4) \
+        (a >= MEM_L1_BASE) && \
+        (a + l <= MEM_L1_BASE + MEM_L1_SIZE) && \
+        ((a) + (l) > (a)) \
     )
 #define DEBUG_VALID_PCIE_ADDR(a, l) (((a) >= NOC_PCIE_ADDR_BASE) && \
                                      ((a) + (l) <= NOC_PCIE_ADDR_END) && \
-                                     ((a) + (l) > (a)) && \
-                                     ((a % NOC_PCIE_ALIGNMENT_BYTES) == 0))
+                                     ((a) + (l) > (a)))
 #define DEBUG_VALID_DRAM_ADDR(a, l) (((a) >= NOC_DRAM_ADDR_BASE) && \
                                      ((a) + (l) <= NOC_DRAM_ADDR_END) && \
-                                     ((a) + (l) > (a)) && \
-                                     ((a % NOC_DRAM_ALIGNMENT_BYTES) == 0))
+                                     ((a) + (l) > (a)))
 
 #define DEBUG_VALID_ETH_ADDR(a, l) (((a) >= MEM_ETH_BASE) && \
-                                    ((a) + (l) <= MEM_ETH_BASE + MEM_ETH_SIZE) && \
-                                    ((a % NOC_L1_ALIGNMENT_BYTES) == 0))
+                                    ((a) + (l) <= MEM_ETH_BASE + MEM_ETH_SIZE))
 
 // Note:
 //  - this isn't racy w/ the host so long as invalid is written last
 //  - this isn't racy between riscvs so long as each gets their own noc_index
-inline void debug_sanitize_post_noc_addr_and_hang(uint64_t a, uint32_t l, uint32_t invalid)
-{
+inline void debug_sanitize_post_noc_addr_and_hang(
+    uint64_t noc_addr,
+    uint32_t l1_addr,
```

Comment:
should we make the "mem_addr" since it isn't necessarily l1?

Reply:
I think this filed specifically refers to the local dst/src address, which unless I'm missing some context should always be L1? Can rename to `local_addr` or similar, will leave it for #6738 when I clean this up a little